fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/KeySpacePath.java

```diff
@@ -432,4 +432,13 @@ default RecordCursor<DataInKeySpacePath> exportAllData(@Nonnull FDBRecordContext
     @Nonnull
     CompletableFuture<Void> importData(@Nonnull FDBRecordContext context,
                                        @Nonnull Iterable<DataInKeySpacePath> dataToImport);
+
+    /**
+     * Two {@link KeySpacePath}s are equal if they have equal values, the same directory (reference equality) and their
+     * parents are the same.
+     * @param obj another {@link KeySpacePath}
+     * @return {@code true} if this path equals {@code obj}
+     */
+    @Override
+    boolean equals(Object obj);
```

Comment:
What's the benefit of adding `equals` to the interface?

Reply:
It adds documentation to how `equals` behaves. You can see similar things on `List` or `Map` in the JDK.